Removal of TypeRepr

spkersten opened the pull request:
More clean up. I've removed the TypeRepr class since it is not used anymore.

spkersten commented on mypy/semanal.py:
```diff
@@ -1944,7 +1944,7 @@ def self_type(typ: TypeInfo) -> Union[Instance, TupleType]:
 def replace_implicit_first_type(sig: Callable, new: Type) -> Callable:
     # We can detect implicit self type by it having no representation.
-    if not sig.arg_types[0].repr:
+    if True: # TODO: repr field was removed
```
This is of course not correct. There is currently not test for it. I'll try to add tests and fix this later.

JukkaL commented:
@spkersten Are you still working on this pull request? I can finish this if you are busy.

spkersten commented:
It has become busy at work, so go ahead if you want to finish it. 


> On 9 feb. 2015, at 01:40, Jukka Lehtosalo <notifications@github.com> wrote:
> 
> @spkersten Are you still working on this pull request? I can finish this if you are busy.
> 
> —
> Reply to this email directly or view it on GitHub.
>

spkersten commented:
I've rebased the branch and removed the 'implicit self type'. As PEP484 specifies that the first argument of a method should not have a type specified, it seems ok to alway replace the type of the first argument by the enclosing class.

JukkaL commented on stubs/3.2/queue.py:
```diff
@@ -5,6 +5,7 @@
 from typing import Any
 
 class Queue():
+    @staticmethod
```
This doesn't look like a static method?

spkersten replied:
Since you merged it, I guess you figured out why I did this. (Type of the first parameter was being changed to `Queue`, because the method was neither static or a class method.)

spkersten replied:
I should have looked more careful in fixing the type error. The `self` parameter is missing here.

spkersten replied:
Should be fixed in #637.
